docker-compose.yml

```diff
@@ -0,0 +1,210 @@
+      timeout: 10s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/dev_appserver.py
+    command: 
+      - "/app/oppia/app_dev_docker.yaml"
+      - "--runtime"
+      - "python38"
+      - "--host=0.0.0.0"
+      - "--port=8181"
+      - "--admin_host=0.0.0.0"
+      - "--admin_port=8000"
+      - "--skip_sdk_update_check=True"
+
+  datastore:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-cloud-datastore
+    ports:
+      - "8089:8089"
+    environment:
+      - CLOUDSDK_CORE_DISABLE_PROMPTS=1
+    depends_on:
+      angular-build:
+        condition: service_healthy
+      webpack-compiler:
+        condition: service_healthy
+    healthcheck:
+      test: curl -f http://localhost:8089/ || exit 1
+      timeout: 10s
+      interval: 2m
+      retries: 5
+      start_period: 20m
+    entrypoint: /google-cloud-sdk/bin/gcloud
+    command:
+      - "beta"
+      - "emulators"
+      - "datastore"
+      - "start"
+      - "--project"
+      - "dev-project-id"
+      - "--data-dir"
+      - "/app/cloud_datastore_emulator_cache"
+      - "--host-port=0.0.0.0:8089"
+      - "--consistency=1.0"
+      - "--quiet"
+      - "--no-store-on-disk"
+
+  firebase:
+    image: node@sha256:48650ae5c5dc9fbd0d0559fd1248e779f7214ba99434420a089bc7853aeefb65
+    container_name: oppia-firebase-emulator
+    ports:
+      - "9099:9099"
+      - "4000:4000"
+    volumes:
+      - ./.firebase.json:/app/oppia/.firebase.json
+      - shared_node_modules:/app/oppia/node_modules
+    environment:
+      - oppia_is_dockerized=true
+    healthcheck:
+      test: wget --spider http://localhost:4000/auth/ && echo 'y' | /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js emulators:export /app/firebase_emulator_cache --project=dev-project-id || exit 1
+      timeout: 20s
+      interval: 20s
+      retries: 5
+    entrypoint:
+      - /app/oppia/node_modules/firebase-tools/lib/bin/firebase.js
+      - "emulators:start"
+      - "--only"
+      - "auth"
+      - "--project=dev-project-id"
+      - "--config=/app/oppia/.firebase.json"
+    command:
+      - "--import=/app/firebase_emulator_cache"
+      - "--export-on-exit=/app/firebase_emulator_cache"
+
+  redis:
+    image: redis@sha256:aebc052b5a7f50d79f00a4deee22109e2ecf22f439140a03f2c9770055ffec7b
+    container_name: oppia-redis
+    ports:
+      - "6379:6379"
+    environment:
+      - PATH=/usr/local/bin/
+    volumes:
+      - oppia_redis_dump:/app/oppia/dump.db
+      - ./redis_docker.conf:/app/oppia/redis_docker.conf
+    healthcheck:
+      test: redis-cli ping
+      timeout: 5s
+      interval: 30s
+      retries: 5
+    command: redis-server /app/oppia/redis_docker.conf
+
+  elasticsearch:
+    image: elasticsearch@sha256:fa7141154a7e14df214e42f08c333702403eb88c02ba44e79322a1f42d733013
+    container_name: oppia-elasticsearch
+    ports:
+      - "9200:9200"
```

Comment:
yes, we require this when accessing any ES services from its APIs. also used within the healthcheck

Reply:
> yes, we require this when accessing any ES services from its APIs.

This will be required by the google app engine docker container and not by the host, right?

Reply:
yes, correct. but this port must be open for even GAE.

Reply:
All the services will be created under one default network, right? ([ref](https://docs.docker.com/compose/networking/)) I think this needs changes but let's discuss this after all the services are created and we can better structure the networks.